From a27a9d06a47cec4f684d7c16d14096f277d4fe30 Mon Sep 17 00:00:00 2001
From: Ricardo Branco <rbranco@suse.de>
Date: Sun, 24 Aug 2025 13:10:17 +0200
Subject: [PATCH] tests: Fix deprecation warning for utcfromtimestamp()

Fix DeprecationWarning for datetime.datetime.utcfromtimestamp()

It suggests to use timezone-aware objects to represent datetimes in UTC
with datetime.UTC but datetime.timezone.utc is backwards compatible.

Signed-off-by: Ricardo Branco <rbranco@suse.de>
---
 podman/api/parse_utils.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/podman/api/parse_utils.py b/podman/api/parse_utils.py
index 39e30da7..0294712d 100644
--- a/podman/api/parse_utils.py
+++ b/podman/api/parse_utils.py
@@ -4,7 +4,7 @@
 import ipaddress
 import json
 import struct
-from datetime import datetime
+from datetime import datetime, timezone
 from typing import Any, Optional, Union
 from collections.abc import Iterator
 
@@ -48,7 +48,9 @@ def prepare_timestamp(value: Union[datetime, int, None]) -> Optional[int]:
         return value
 
     if isinstance(value, datetime):
-        delta = value - datetime.utcfromtimestamp(0)
+        if value.tzinfo is None:
+            value = value.replace(tzinfo=timezone.utc)
+        delta = value - datetime.fromtimestamp(0, timezone.utc)
         return delta.seconds + delta.days * 24 * 3600
 
     raise ValueError(f"Type '{type(value)}' is not supported by prepare_timestamp()")
